src/Compilers/CSharp/Portable/Symbols/PublicModel/TypeSymbol.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal abstract class TypeSymbol : NamespaceOrTypeSymbol, ISymbol, ITypeSymbol
+    {
+        protected TypeSymbol(CodeAnalysis.NullableAnnotation nullableAnnotation)
+        {
+            NullableAnnotation = nullableAnnotation;
+        }
+
+        protected CodeAnalysis.NullableAnnotation NullableAnnotation { get; }
+
+        protected abstract ITypeSymbol WithNullableAnnotation(CodeAnalysis.NullableAnnotation nullableAnnotation);
+
+        internal abstract Symbols.TypeSymbol UnderlyingTypeSymbol { get; }
+
+        CodeAnalysis.NullableAnnotation ITypeSymbol.NullableAnnotation => NullableAnnotation;
+
+        ITypeSymbol ITypeSymbol.WithNullableAnnotation(CodeAnalysis.NullableAnnotation nullableAnnotation)
+        {
+            if (NullableAnnotation == nullableAnnotation)
+            {
+                return this;
+            }
+            else if (nullableAnnotation == UnderlyingTypeSymbol.DefaultNullableAnnotation)
+            {
+                return (TypeSymbol)UnderlyingSymbol.ISymbol;
```

Comment:
>TypeSymbol [](start = 24, length = 10)

`ITypeSymbol`?

Reply:
> ITypeSymbol?

The code is actually correct because this ```TypeSymbol``` is ```Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.TypeSymbol``` (the enclosing class), which implements ```ITypeSymbol``` and has an implicit conversion to ```ITypeSymbol```. But I agree, this might be confusing to a reader.

---
In reply to: [338830978](https://github.com/dotnet/roslyn/pull/39498#discussion_r338830978) [](ancestors = 338830978)